de.dlr.sc.overtarget.language.ui/src/de/dlr/sc/overtarget/language/ui/quickfix/OvertargetQuickfixProvider.xtend

```diff
@@ -82,4 +88,45 @@ class OvertargetQuickfixProvider extends DefaultQuickfixProvider {
 			}
 		}, 1)
 	}
+	
+	@Inject
+	IResourceSetProvider resourceSetProvider
+	
+	@Fix(OvertargetValidator.FILE_NAME_LIKE_TARGET_NAME)
+	def fixFileNameLikeTargetName(Issue issue, IssueResolutionAcceptor acceptor) {
+		acceptor.accept(issue, 'Replace with correct tmodel name', '', 'upcase.png') [
+		
+		context |
+			val editor = PlatformUI.getWorkbench().getActiveWorkbenchWindow().getActivePage().getActiveEditor()
+			if (editor instanceof ITextEditor) {
+				val progressMonitor = new NullProgressMonitor()
+				editor.doSave(progressMonitor) //saves the made changes in the file
+				val ite = editor as ITextEditor
+				val input = ite.editorInput
+				val fileEditorInput = input as IFileEditorInput
+				val file = fileEditorInput.file
+				val uri = URI.createPlatformResourceURI(file.getFullPath().toString(), true);
+				val project = file.getProject();
+				val rs = resourceSetProvider.get(project);
+				val r = rs.getResource(uri, true);
+				
+				val model = r.contents.get(0) as TargetFile
+				val modelName = model.name
+				
+				val targetKeyword = grammarAccess.KEYWORDAccess.targetKeyword_1.value
+				val targetLibraryKeyword = grammarAccess.targetLibraryRule.name
+				val WHITESPACE_SEPARATOR = 1
+				
+				val xtextDocument = context.xtextDocument
+				val fileName = editor.title.replace(".tmodel", "")
```

Comment:
Sure it's a good idea to use the editor title to get the filename? Maybe xtext descides to change the editor title to "XtextEditor: My.tmodel" with the next version... I'd prefer using the actually filename directly

Reply:
Yes, me too